src/mcts/node.cc

```diff
@@ -113,9 +113,55 @@ Move Edge::GetMove(bool as_opponent) const {
   return m;
 }
 
+// Policy priors (P) are stored in a compressed 16-bit format.
+//
+// Source values are 32-bit floats:
+// * bit 31 is sign (zero means positive)
+// * bit 30 is sign of exponent (zero means nonpositive)
+// * bits 29..23 are value bits of exponent
+// * bits 22..0 are significand bits (plus a "virtual" always-on bit: s ∈ [1,2))
+// The number is then sign * 2^exponent * significand, usually.
+// See https://www.h-schmidt.net/FloatConverter/IEEE754.html for details.
+//
+// In compressed 16-bit value we store bits 27..12:
+// * bit 31 is always off as values are always >= 0
+// * bit 30 is always off as values are always < 2
+// * bits 29..28 are only off for values < 4.6566e-10, assume they are always on
+// * bits 11..0 are for higher precision, they are dropped leaving only 11 bits
+//     of precision
+//
+// When converting to compressed format, bit 11 is added to in order to make it
+// a rounding rather than truncation.
+//
+// Out of 65556 possible values, 2047 are outside of [0,1] interval (they are in
+// interval (1,2)). This is fine because the values in [0,1] are skewed towards
+// 0, which is also exactly how the components of policy tend to behave (since
+// they add up to 1).
+
+// If the two assumed-on exponent bits (3<<28) are in fact off, the input is
+// rounded up to the smallest value with them on. We verify this by subtracting
+// the two bits from the input and checking for a negative result. This is
+// combined with the round-to-nearest addition (1<<11).
+void Edge::SetP(float p) {
+  constexpr int32_t roundings = (1 << 11) - (3 << 28);
+  assert(0.0f <= p && p <= 1.0f);
+  int32_t tmp;
+  std::memcpy(&tmp, &p, sizeof(float));
+  tmp += roundings;
```

Comment:
it was `-=` prior to change, is `+=` needed instead?

Reply:
Notice I flipped the order of the two constants as well, it's equivalent either way.